### PR TITLE
AI rework for destination and minmax

### PR DESCRIPTION
**Pull request by BWalusiak**

### CODE REVIEW

File: app/src/main/java/com/cyberbot/checkers/game/Grid.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package com.cyberbot.checkers.game;
 
+import android.net.wifi.p2p.WifiP2pManager;
```
Comment (RouNNdeL) [inline app/src/main/java/com/cyberbot/checkers/game/Grid.java]: No Android related code in game package. Looks like an unused import.

File: app/src/main/java/com/cyberbot/checkers/game/Grid.java [hunk 2]
```diff
@@ -369,34 +370,39 @@ int getValue(PlayerNum playerNum, PlayerNum adversaryNum) {
...
             }
         }
-        return value;
+        return enemyNo == 0;
     }
 
-    static Grid simulateMove(Grid grid, Move move) {
-        grid.attemptMove(move.getStart(), move.getEnd());
-        return grid;
+    boolean loose(PlayerNum player){
```
Comment (RouNNdeL) [inline app/src/main/java/com/cyberbot/checkers/game/Grid.java]: Should probably rename to `didLose` or `lost`, `loose` (not only being a spelling error) would reflect that it makes the player lose.

Comment (RouNNdeL) [reply app/src/main/java/com/cyberbot/checkers/game/Grid.java]: It should also check if the player has moves available. There might still be pieces on the board despite no moves left.

**Review by RouNNdeL — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/cyberbot/checkers/game/Grid.java (hunk 1 above)
Comment (BWalusiak) [reply app/src/main/java/com/cyberbot/checkers/game/Grid.java]: removed

File: app/src/main/java/com/cyberbot/checkers/game/Grid.java (hunk 2 above)
Comment (BWalusiak) [reply app/src/main/java/com/cyberbot/checkers/game/Grid.java]: renamed to lost and won, move available check added

**Review by RouNNdeL — APPROVED** [review]

**Pull request merged**